### DISCUSSION TITLE
`Forwarded` header access improvement

### DISCUSSION THREAD

**Discussion by kellytk** [tokio-rs/axum#462] [Ideas]
My web server and reverse proxy is configured to set the `Forwarded` header, which is the standardized replacement for the `X-Forwarded-*` headers.

Using the `HeaderMap` extractor with `.get(FORWARDED)` still leaves a string to parse.  A more structured means of accessing its data would be helpful.  Should I open a feature request issue in this repo, that of [`headers`](https://github.com/hyperium/headers), [`http`](https://github.com/hyperium/http), or elsewhere?

**Comment by davidpdrsn** [comment]
`headers` sounds like the right place for that.

**Reply by kellytk** [reply]
[Opened](https://github.com/hyperium/headers/issues/96).

Until it's added, and accessing `Forwarded` via `TypedHeader` is supported, is the best alternative to use the `HeaderMap` extractor?

**Reply by davidpdrsn** [reply]
Writing your own extractor.
